### ISSUE TITLE
Extend `Get-IshTypeFieldDefinition` to understand IMetadataBinding fields avoiding errors like "Field has configured metadata binding and it's value type can only be 'id' or 'element'"

### ISSUE THREAD

**Issue by ddemeyer** [sdl/ISHRemote#58]
The problem is errors like `[-105006] Value type "none" is not allowed on field "FMBCONTENTREFTYPE". Field has configured metadata binding and it's value type can only be 'id' or 'element' [f:16244 fe:FMBCONTENTREFTYPE ft:Text] [105006;InvalidValueTypeForFieldType]` which can happen on `Add-IshDocumentObj` or `Set-IshDocumentObj` and variations for PublicationOutput, etc

The problem comes that an initialized IshObject contains the _value_ for the metadatabounded fields, but the API doesn't allow those in because 'value' is simply not unique enough to integrate systems, complaining with only 'id' or 'element' allowed. `IshTypeFieldSetup::ToIshMetadataFields` doesn't know metadatabinding fields, so just leaves them in; while it should actually filter those

Knowing that Xml Setup field `FISHEXTENSIONCONFIG` contains the metadatabinding configuration
```
<infoShareExtensionConfig version="1.0">
  <metadatabindings>
  <metadatabinding ishfieldname="FMBPRODUCTFAMILYNAME" sourceref="SitesTaxonomyFamilyName" />
  <metadatabinding ishfieldname="FMBPRODUCTRELEASENAME" sourceref="SitesTaxonomyReleaseName" />
  <metadatabinding ishfieldname="FMBCONTENTREFTYPE" sourceref="SitesTaxonomyContentReferenceType" />	
</metadatabindings>
...
```

- [x] Just in time retrieval on `IshSession` of an *internal* `IshExtensionConfig`. use the root `@version` to decide how to parse. Make sure to add a test on `NewIshSession.test.ps1` or `GetIshTypeFieldDefinition.test.ps1`
- [x] Extend object `IshTypeFieldSetup` to use `IshExtensionConfig`, especially in `ToIshMetadataFields` by filtering out valuetype _value_ for fields configured in metadatabindings
- [x] Extend `Get-IshTypeFieldDefinition` to have `ReferenceMetadataBinding` next to existing `ReferenceLov` and `ReferenceType`, nicely rendered under one column via `ISHRemote.Format.ps1xml`
- [x] Double check `AddIshDocumentObj.Tests.ps1` lines 152, 163, 171 and 182 on a mashup-enabled system

**Comment by ddemeyer** [comment]
1. Issue #58 explains how to implement ISHMetadataBinding visibility in `Get-IshTypeFieldDefinition` across all supported versions. Starting in 14SP4/14.0.4, the Settings25 API call will be extended, see issue #106 so make sure to align implementation between these two issues.
1. So after implementation of #106, issue #58 serves as backfilling for the _older_ versions so `Get-IshTypeFieldDefinition` works the same on any version by explicitly getting and parsing the extension configuration. 
1. Next the "Field has configured metadata binding and it's value type can only be 'id' or 'element'" bug can be fixed, as the field can now be filtered on a write operation. Also covered under #58

**Comment by ddemeyer** [comment]
**Short...**
Added static IshSettingsExtensionConfig::MergeIntoIshTypeFieldSetup to pimp IshFieldSetup for the 13/13.0.0 up until 14SP4/14.0.4 legacy case, starting 14SP4/14.0.4 the TypeFieldSetup API call handles this implicitly. Extended Get-IshTypeFieldDefinition and (New-)IshSession. Also verifed Compare-IshTypeFIeldDefinition. Thanks to #106, issue #58 is solved on version 13/13.0.0 and above.

**Longer...**
Decided to not introduce an `IshSettingsExtensionConfig.cs` object yet, just the CS file holding a static method to merge/pimp IshTypeFieldSetup with a big try-catch. This in turn means you cannot easily run a test. After all #58 is about backward compatibility of newly introduced API function of #106 for versions 13/13.0.0 up until 14SP4/14.0.4

Verified
* Ran Pester tests on 14SP3/14.0.3 (GA+CUHotfix) without any metadatabinding configured and on 15-ALPHA holding 3+ connectors with success
* Manual test of `New-IshSession` and `Get-IshTypeFieldDefinition` on versions 13SP1, 14SP2, 14SP3, 14SP4-ALPHA, 15-ALPHA
* `Compare-IshTypeFieldDefinition -LeftIshSession $left -RightIshSession $right` is implicit as it relies on the above two for initialization of `IshTypeFieldSetup` or `IshTypeFieldDefinition[]`

And also ran the below, which initially failed with an error as described above, and now works.
```
New-IshSession -WsBaseUrl 	https://...cdev14qa04.../ISHWSORA19/ -PSCredential admin -Debug -Verbose
$o= (Get-IshDocumentObj -LogicalId ISHEDITORTEMPLATEMODULETOPIC)[-1]  # -1 pulls last entry from the array
$o = Find-IshDocumentObj -MetadataFilter (Set-IshMetadataFilterField -Level Logical -Name FTESTCONTINENTS -ValueType Element -FilterOperator NotEmpty)
$o.ftestcontinents_logical_value # returning Europe, Asia
$o | Set-IshDocumentObj
```
